src/debugSession/BrightScriptDebugSession.ts

```diff
@@ -496,7 +496,17 @@ export class BrightScriptDebugSession extends BaseDebugSession {
             failOnCompileError: true
         }).then(() => {
             packageIsPublished = true;
-        }).catch((e) => {
+        }).catch(async (e) => {
+            const statusCode = e?.results?.response?.statusCode;
+            const message = e.message as string;
+            if (statusCode !== 200) {
+                this.showPopupMessage(message, 'error', true);
+                if (connectPromise !== undefined) {
+                    await this.shutdown(message);
+                }
+
+                throw new Error(message);
```

Comment:
I think you can rethrow the same error rather than making a new one. 
```suggestion
                throw e;
```